internal/v2/s2av2.go

```diff
@@ -0,0 +1,72 @@
+package s2av2
+
+import (
+	"errors"
+	"context"
+
+	"google.golang.org/grpc/credentials"
+	"github.com/google/s2a-go/internal/v2/tlsconfigstore"
+	"github.com/google/s2a-go/internal/handshaker"
+)
+
+const (
+	s2aSecurityProtocol = "s2av2"
+)
+
+type s2av2TransportCreds struct {
+	info     *credentials.ProtocolInfo
+	config   *tls.Config
+	isClient bool
+}
+
+func NewClientCreds() (*credentials.TransportCredentials) {
```

Comment:
nit: I would suggest removing the `*` in front of `credentials.TransportCredentials`. Generally you return an interface directly rather than a pointer to an interface (but you can return a pointer to an instance of a struct, e.g.).